enterprise/views.py

```diff
@@ -248,10 +248,21 @@ def post_account_consent(self, request, consent_provided):
         Interpret the account-wide form above, and save it to a UserDataSharingConsentAudit object for later retrieval.
         """
         self.lift_quarantine(request)
+
         customer = get_enterprise_customer_for_request(request)
-        if customer is None:
+        social_auth = get_real_social_auth_object(request)
+        user = getattr(social_auth, 'user', None)
+        if customer is None or user is None:
+            # Check that we can get both a user and an EnterpriseCustomer from the running
+            # TPA pipeline; if not, this view isn't relevant, and shouldn't be used.
             raise Http404
-        user = get_real_social_auth_object(request).user
+
+        if not consent_provided and active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN):
```

Comment:
Does it make sense to perform this check prior to loading the customer+user information above?